### PR TITLE
[SPARK-35938][PYTHON] Add deprecation warning for Python 3.6

### PR DESCRIPTION
**Pull request by xinrong-meng**
### What changes were proposed in this pull request?

Add deprecation warning for Python 3.6.

### Why are the changes needed?

According to https://endoflife.date/python, Python 3.6 will be EOL on 23 Dec, 2021.
We should prepare for the deprecation of Python 3.6 support in Spark in advance.

### Does this PR introduce _any_ user-facing change?

N/A.

### How was this patch tested?

Manual tests.

### CODE REVIEW

**Comment by dongjoon-hyun** [comment]
Thank you for working on this, @xinrong-databricks .

cc @HyukjinKwon

File: python/pyspark/context.py [hunk 1]
```diff
@@ -230,6 +230,14 @@ def _do_init(self, master, appName, sparkHome, pyFiles, environment, batchSize,
         self.pythonExec = os.environ.get("PYSPARK_PYTHON", 'python3')
         self.pythonVer = "%d.%d" % sys.version_info[:2]
 
+        if self.pythonVer == (3, 6):
```
Comment (ueshin) [inline python/pyspark/context.py:233]: nit: I guess `self.pythonVer < (3, 7)` is better?

File: python/pyspark/context.py [hunk 2]
```diff
@@ -230,6 +230,14 @@ def _do_init(self, master, appName, sparkHome, pyFiles, environment, batchSize,
...
+            with warnings.catch_warnings():
+                warnings.simplefilter("once")
+                warnings.warn(
+                    "Support for Python 3.6 is deprecated as of Spark 3.2.",
+                    DeprecationWarning
```
Comment (ueshin) [inline python/pyspark/context.py:238]: `FutureWarning`?

File: python/pyspark/context.py (hunk 1 above)
Comment (xinrong-databricks) [reply python/pyspark/context.py]: Cool, can we still warn `Python 3.6` only in the message?

File: python/pyspark/context.py (hunk 2 above)
Comment (xinrong-databricks) [reply python/pyspark/context.py]: Makes sense!

File: python/pyspark/context.py (hunk 1 above)
Comment (ueshin) [reply python/pyspark/context.py]: Yes, that's fine.

File: python/pyspark/context.py [hunk 3]
```diff
@@ -230,6 +230,14 @@ def _do_init(self, master, appName, sparkHome, pyFiles, environment, batchSize,
         self.pythonExec = os.environ.get("PYSPARK_PYTHON", 'python3')
         self.pythonVer = "%d.%d" % sys.version_info[:2]
 
+        if self.pythonVer < (3, 7):
+            with warnings.catch_warnings():
+                warnings.simplefilter("once")
+                warnings.warn(
+                    "Python 3.6 support will be deprecated in Spark 3.2.",
```
Comment (ueshin) [inline python/pyspark/context.py:237]: `Python 3.6 support is deprecated in Spark 3.2.`?

File: python/pyspark/context.py (hunk 1 above)
Comment (ueshin) [reply python/pyspark/context.py]: ah, sorry, it should be `sys.version_info < (3, 7)`.

File: python/pyspark/context.py (hunk 3 above)
Comment (xinrong-databricks) [reply python/pyspark/context.py]: Cool.

**Review by ueshin — APPROVED** [review]

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**